authentication/oauth.go

```diff
@@ -62,6 +62,123 @@ func (o *OAuth) LoginWithPassword(ctx context.Context, body oauth.LoginWithPassw
 	return
 }
 
+// LoginWithAuthCode performs the Authorization Code grant type OAuth 2.0 grant.
+//
+// This is the flow that regular web apps use to access an API. Use this endpoint to exchange an
+// Authorization Code for a Token.
+//
+// See: https://auth0.com/docs/api/authentication?http#authorization-code-flow44
+func (o *OAuth) LoginWithAuthCode(ctx context.Context, body oauth.LoginWithAuthCodeRequest, opts ...RequestOption) (t *oauth.TokenSet, err error) {
+	data := url.Values{
+		"code": []string{body.Code},
+	}
+
+	if body.RedirectURI != "" {
+		data.Set("redirect_uri", body.RedirectURI)
+	}
+
+	err = o.addClientAuthentication(body.ClientAuthentication, data, true)
+
+	if err != nil {
+		return
+	}
+
+	t, err = o.LoginWithGrant(ctx, "authorization_code", data, opts...)
+	return
+}
+
+// LoginWithAuthCodeWithPKCE performs the Authorization Code with Proof Key for Code Exchange OAuth 2.0 grant type.
+//
+// This flow was originally designed to protect the authorization code flow in mobile apps but its ability
+// to prevent authorization code injection makes it useful for every type of OAuth client, even web apps
+// that use client authentication.
+//
+// See: https://auth0.com/docs/api/authentication?http#authorization-code-flow-with-pkce45
+func (o *OAuth) LoginWithAuthCodeWithPKCE(ctx context.Context, body oauth.LoginWithAuthCodeWithPKCERequest, opts ...RequestOption) (t *oauth.TokenSet, err error) {
+	data := url.Values{
+		"code":          []string{body.Code},
+		"code_verifier": []string{body.CodeVerifier},
+	}
+
+	if body.RedirectURI != "" {
+		data.Set("redirect_uri", body.RedirectURI)
+	}
+
+	err = o.addClientAuthentication(body.ClientAuthentication, data, false)
+
+	if err != nil {
+		return
+	}
+
+	t, err = o.LoginWithGrant(ctx, "authorization_code", data, opts...)
+	return
+}
+
+// LoginWithClientCredentials performs the Client Credentials OAuth 2.0 grant type.
+//
+// This flow was originally designed to protect the authorization code flow in mobile apps but its ability
+// to prevent authorization code injection makes it useful for every type of OAuth client, even web apps
+// that use client authentication.
+//
+// See: https://auth0.com/docs/api/authentication?http#client-credentials-flow
+func (o *OAuth) LoginWithClientCredentials(ctx context.Context, body oauth.LoginWithClientCredentialsRequest, opts ...RequestOption) (t *oauth.TokenSet, err error) {
+	data := url.Values{
+		"code": []string{body.Audience},
+	}
+
+	err = o.addClientAuthentication(body.ClientAuthentication, data, true)
+
+	if err != nil {
+		return
+	}
+
+	t, err = o.LoginWithGrant(ctx, "client_credentials", data, opts...)
+	return
+}
+
+// RefreshToken is used to refresh and Access Token using the Refresh Token you got during authorization.
```

Comment:
```suggestion
// RefreshToken is used to refresh and access token using the refresh token you got during authorization.
```
The docs team's guidelines instruct not to title-case industry terms like "access token" and "refresh token".